query-planner-js/src/buildPlan.ts

```diff
@@ -256,6 +257,21 @@ class QueryPlanningTaversal<RV extends Vertex> {
     return FetchDependencyGraph.create(this.subgraphs);
   }
 
+  // Moves the first closed branch to after any branch having more options.
+  private reorderFirstBranch() {
+    const firstBranch = this.closedBranches[0];
+    let i = 1;
+    while (i < this.closedBranches.length && this.closedBranches[i].length > firstBranch.length) {
```

Comment:
I'm not sure if this is not working or if I just don't understand what is desired.
```text
Suppose [4,2,7,9]. In that case i will stay at 1 resulting in [2,4,7,9]
Suppose [2,5,4,7]. In this case we'll increment to the end and swap, resulting in [7,5,4,2]
Suppose [2,5,1,7]. In this case we'll increment until we find `1` and swap, resulting in [1,5,2,7]
```
That's the behavior we want?

Also I think that if you have an array of two elements, they will be swapped regardless of what the values are. That seems wrong.

Reply:
Oh, missed your comment that said it's already sorted. I still think there is an issue where it'll always swap even if it shouldn't.

[9,8] -> [8,9]

Reply:
> missed your comment that said it's already sorted

Yeah, added a comment to the method itself as that wasn't very clear.

> I still think there is an issue where it'll always swap even if it shouldn't.

Is it? If the input is [9, 8] (which I'm interpreting as a list of length 9 followed by one of size 8), then the very first test of the while loop would fail (the 2nd test is then `8 > 9` which is false), and the post-loop code would just re-assign the first element to itself, so not swap. Am I misunderstood the case you are referring to?

Reply:
My bad, didn't notice you were swapping with `i-1` instead of `i`